frontend/catalyst/pennylane_extensions.py

```diff
@@ -1379,6 +1390,94 @@ def _decorator(true_fn: Callable):
     return _decorator
 
 
+class ForLoopCallable:
```

Comment:
Why do we need this new class and where is it called? I see that it follows the same model as class `CondCallable`

Reply:
This is actually what we used to have. It is an alternative to the "three-tiered" decorator:
```py
def for_loop(start, stop, step):
   def decorator(fn):
      def wrapper(*args):
         ...
      return wrapper
   return decorator
```
but allows us to maintain state on the wrapped object as well as add methods like `set_id` and `otherwise` (on the `cond` decorator). The user invokes this class when they "call" the for loop.